### PR TITLE
New Web Application

### PR DESCRIPTION
**Pull request by francescomalatesta**
Here's the new version of the ASTEF web application.

You will find the instructions to use it in the `src/webapp/README.md` file.

For my tests, I actually I used the [flightdeck-fix.txt](https://github.com/theskinnerbox/ASTEF/blob/master/demo_data/flightdeck-fix.txt) file. 

The only thing that remains left to implement is the background image upload for the chart, but I am experiencing some issues with it. I will update you ASAP.

Have a nice weekend! :)

EDIT: don't merge it right now, let's test it first with other fixations txt files.

EDIT2: live demo @ [http://astef-web.netlify.com/](http://astef-web.netlify.com/) (it's a cheap free hosting, so it could take a while to load :D)

### CODE REVIEW

**Comment by wbigger** [comment]
Well done! Some notes:
- start the minute counting from one instead of zero
- add a "Save time series" button that saves a file with NNI values as in the matlab version:
  [flightdeck-fix-nni.txt](https://github.com/theskinnerbox/ASTEF/files/331979/flightdeck-fix-nni.txt)

**Comment by francescomalatesta** [comment]
More notes from @dinocera :
- change the color of the current NNI value;
- upload a background picture for the scanpath;
- instead of "Per Minute Fixations", use "Scanpath";

**Comment by francescomalatesta** [comment]
Currently done:
- start the minute counting from one instead of zero
- add a "Save time series" button that saves a file with NNI values as in the matlab version: flightdeck-fix-nni.txt
- instead of "Per Minute Fixations", use "Scanpath";

The remaining ones,
- change the color of the current NNI value;
- upload a background picture for the scanpath;

seems to be a problem for our charts library. I'll let you know if I can figure out something :)

File: src/webapp/src/Processors/FixationsDataProcessor.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            this.fixationsData[minute] = {
+                points: [point]
+            }
+        } else {
+            this.fixationsData[minute].points.push(point);
+        }
+    }
+
+    getMinuteFromTimestamp(timestamp) {
+        let minute = Math.floor(timestamp * 1.67 * 0.00001);
```
Comment (wbigger) [inline src/webapp/src/Processors/FixationsDataProcessor.js]: Questa formula è approssimata e porta a risultati diversi da quelli di matlab! Usare una formula più precisa, ad esempio
`let minute = Math.floor(timestamp / 1000 / 60);`

**Review by wbigger — CHANGES_REQUESTED** [review]
Ciao Francesco, ho fatto dei test e c'è una modifica da fare sulla suddivisione delle fissazioni per minuto.

File: src/webapp/src/Processors/FixationsDataProcessor.js (hunk 1 above)
Comment (francescomalatesta) [reply src/webapp/src/Processors/FixationsDataProcessor.js]: Sistemato!

**Review by wbigger — APPROVED** [review]

**Comment by wbigger** [comment]
Issues:
- [x] change the color of the current NNI value
- [ ] upload a background picture for the scanpath;

The image is correctly loaded but not properly resized. The whole image should be resized to fit the drawing area. Indeed, the fixations has to exactly overlay the background image.

Following screenshots use the jpg image in demo_data folder.

Correct example, from Matlab implementation:
screen shot 2017-01-23 at 10 37 10 am

Current implementation in my browser (Chrome 55.0.2883.95 on macOS Sierra):
screen shot 2017-01-23 at 10 36 59 am

**Comment by francescomalatesta** [comment]
@wbigger this seems very strange, actually I am using Chrome 52 on Ubuntu and I am not experiencing any issues. However, try again now, it should be fixed as I fixed the resize event. Tomorrow I will also try it at office.

**Comment by wbigger** [comment]
@francescomalatesta still the same issue with the last version :(

**Comment by francescomalatesta** [comment]
@wbigger so strange :( will try tomorrow with a friend's Mac, will keep you updated!

**Review by wbigger — APPROVED** [review]
Still to fix a resize in background. Approving to split this repo into several repo, each one language-specific. The next changes will be made with more atomic fixes!

**Pull request merged**